rust/gitxetcore/src/command/login.rs

```diff
@@ -82,10 +87,167 @@ fn apply_config(
+            "Unable to run aws cli".to_string(),
+        ));
+    }
+    let output = Command::new("aws")
+        .args([
+            "configure",
+            "--profile",
+            &profilename,
+            "set",
+            "aws_secret_access_key",
+            &aws_secret_key,
+        ])
+        .status()?;
+    if !output.success() {
+        return Err(errors::GitXetRepoError::Other(
+            "Unable to run aws cli".to_string(),
+        ));
+    }
+    Ok(profilename.to_string())
+}
+
+/// Prints the AWS Config stdout.
+/// Does not do anything if there is no user config.
+fn print_s3_config(cfg: &Cfg) -> errors::Result<()> {
+    if let Some(ref user) = cfg.user {
+        let aws_access_key =
+            &user
+                .aws_access_key
+                .as_ref()
+                .ok_or(errors::GitXetRepoError::AuthError(anyhow!(
+                    "Not authenticated. Run git-xet login first"
+                )))?;
+        let aws_secret_key =
+            &user
+                .aws_secret_key
+                .as_ref()
+                .ok_or(errors::GitXetRepoError::AuthError(anyhow!(
+                    "Not authenticated. Run git-xet login first"
+                )))?;
+        println!("\tAWS_ACCESS_KEY_ID = {aws_access_key}");
+        println!("\tAWS_SECRET_ACCESS_KEY = {aws_secret_key}");
+    }
+    Ok(())
+}
+
+/// handles the --s3 option and handles all the printing
+/// Eats all errors. Does not return errors.
+fn handle_s3_login_option(cfg: &Cfg, host: &str, try_write_aws_config: bool) {
+    if cfg.user.is_none() {
+        eprintln!("No user configuration. Not authenticated");
+        return;
+    }
+    let user = cfg.user.as_ref().unwrap();
+    if user.aws_access_key.is_none() || user.aws_secret_key.is_none() {
+        eprintln!(
+            "AWS configuration not found.\n\
+            Xethub service failed to provide credentials.\n\
+            Please contact support or your administrator"
+        );
+        return;
+    }
+
+    let mut failed_to_run_awscli: bool = false;
+    let maybe_profilename: Option<String>;
+    let ok;
+    if try_write_aws_config {
+        maybe_profilename = write_aws_config(cfg, host).ok();
+        ok = maybe_profilename.is_some();
+        if !ok {
+            failed_to_run_awscli = true;
+            // actually failed to run aws.
+            eprintln!(
+                "Failed to run aws cli.\n\
+              Unable to configure S3 automatically."
+            );
+        }
+    } else {
+        maybe_profilename = None;
+        ok = false;
+    }
+    if !ok {
+        // instructions without an aws profile
+        eprintln!("Set the following environment variables to use AWS cli");
+        if !failed_to_run_awscli {
+            eprintln!("Re-run `git-xet login` with the --s3 option to write these keys to an AWS cli configuration profile")
+        }
+        eprintln!("");
+        let _ = print_s3_config(cfg);
```

Comment:
Trying to trace out the path when things are not set up well and it seems like the error could stack up on each other without making sense:  E.g.:

Failed to run aws cli.
Unable to configure S3 automatically.
Not authenticated. Run git-xet login first.
To see the repos you have access to via S3,
set the environment variables above and run:
    aws --endpoint-url=s3.xethub.com s3 ls

Reply:
The errors are all eaten and not printed. (let _ = print_s3_config)

Reply:
These are the eprintln! statements, not the errors.